src/Redemption.js

```diff
@@ -133,11 +133,31 @@ export default class Redemption {
     )
   }
 
+  /**
+   * @typedef {Object} AutoSubmitState
+   * @prop {Promise<BitcoinTransaction>} signedTransaction
+   * @prop {Promise<{ transaction: FoundTransaction, requiredConfirmations: Number }>} confirmations
+   * @prop {Promise<EthereumTransaction>} proofTransaction
+   */
+  /**
+   * This method enables the redemption's auto-submission capabilities.
+   *
+   * Calling this function more than once will return the existing state of
+   * the first auto submission process, rather than restarting the process.
+   *
+   * @return {AutoSubmitState} An object with promises to various stages of
+   *         the auto-submit lifetime. Each promise can be fulfilled or
+   *         rejected, and they are in a sequence where later promises will be
+   *         rejected by earlier ones.
+   */
   autoSubmit() {
-    if (this.autoSubmitPromise) {
-      return this.autoSubmitPromise
+    if (this.autoSubmittingState) {
+      return this.autoSubmittingState
     }
-    this.autoSubmitPromise = this.signedTransaction.then(
+
+    const state = (this.autoSubmittingState = {})
+
+    state.signedTransaction = this.signedTransaction.then(
```

Comment:
I don't think this is quite right -> this is the broadcast transaction, not the signed transaction, and all it includes is the `transactionID` (if we get it from `findScript` it's actually a `FoundTransaction`, but if it comes from `broadcast` it isn't, so `transactionID` is all we can rely on). I think we can just rename to `broadcastTransaction` and be fine, or we can call it `broadcastTransactionID` and pull the ID out of the transaction when we `return` below.

Reply:
You're right; we already got the signed transaction. Updated!